vms/platformvm/state/state.go

```diff
@@ -330,13 +339,15 @@ type state struct {
 	expiryDiff *expiryDiff
 	expiryDB   database.Database
 
-	activeSOVLookup        map[ids.ID]SubnetOnlyValidator
-	activeSOVs             *btree.BTreeG[SubnetOnlyValidator]
+	activeSOVs             *activeSubnetOnlyValidators
 	sovDiff                *subnetOnlyValidatorsDiff
 	subnetOnlyValidatorsDB database.Database
+	weightsCache           cache.Cacher[ids.ID, uint64] // subnetID -> total SoV weight
 	weightsDB              database.Database
+	subnetIDNodeIDCache    cache.Cacher[subnetIDNodeID, bool] // subnetID+nodeID -> is validator
```

Comment:
I'm not sure that it actually helps for our planned usage of the `validationID`. We would need to be performing `Get`s on the `subnetIDNodeIDDB`, but I think right now we are planning on iterating over a `subnetID` prefix. The iteration provides the values along with the keys, so I don't think we will actually ever be able to use the cached `validationID`s.

If it would be useful later, then we can definitely change it then... But for now I felt like it didn't make sense.